service/src/main/java/bio/terra/landingzone/stairway/flight/create/CreateLandingZoneFlight.java

```diff
@@ -40,16 +41,37 @@ private void addCreateSteps(LandingZoneFlightBeanBag flightBeanBag, FlightMap in
       throw new LandingZoneCreateException("Unable to find requested landing zone in input map");
     }
 
+    var landingZoneId = inputParameters.get(LandingZoneFlightMapKeys.LANDING_ZONE_ID, UUID.class);
+    if (landingZoneId == null) {
+      throw new LandingZoneCreateException("Landing zone identifier is not defined.");
+    }
+
     addStep(
         new CreateSamResourceStep(flightBeanBag.getSamService()), RetryRules.shortExponential());
 
     addStep(
         new GetBillingProfileStep(flightBeanBag.getBpmService()), RetryRules.shortExponential());
 
     if (!requestedLandingZone.isAttaching()) {
-      addStep(
-          new CreateAzureLandingZoneStep(flightBeanBag.getAzureLandingZoneManagerProvider()),
-          RetryRules.cloud());
+      // TODO: use step which run sub-flight with all these steps
+      // so, in this case we need only one step to rollback everything at once
+      // and don't need to implement rollback for each step
+      // Which option is better here 1 rollback or rollback per resource creation?
+      if (Boolean.TRUE.equals(requestedLandingZone.stairwayPath())) {
```

Comment:
This is what we were talking about yesterday right? So consumers can pass `stairwayPath == false` and get the "legacy" code path? I like this pattern a lot, fwiw so can iterate quickly on the new codepath and not block downstream efforts like adding the needed diagnostic settings etc.

Reply:
Yes. My initial intent was to have simple way of switching between different LZ creation mechanisms and be able to switch to non-stairway functionality for testing purposes and not to block new efforts for current functionality.